src/planning.py

```diff
@@ -423,6 +424,52 @@ def _run_low_level_search(task: Task, option_model: _OptionModelBase,
     return [], True
 
 
+def _run_plan_with_option_model(
+        task: Task, task_idx: int, option_model: _OptionModelBase,
+        plan: List[_Option]) -> Tuple[LowLevelTrajectory, bool]:
+    """Runs a plan on an option model to generate a low level trajectory.
+
+    Returns a LowLevelTrajectory and a boolean. If the boolean is True,
+    the option sequence successfully executed to achieve the goal and
+    generated a LowLevelTrajectory. Otherwise, it returns an empty list
+    and False.
+    """
+    traj: List[State] = [task.init] + [DefaultState for _ in plan]
+    actions: List[Action] = [Action(np.array([0.0])) for _ in plan]
+    for idx in range(len(plan)):
+        state = traj[idx]
+        option = plan[idx]
+        if not option.initiable(state):
+            # The option is not initiable.
+            return LowLevelTrajectory([task.init], [], True, task_idx), False
+        next_state, _ = option_model.get_next_state_and_num_actions(
+            state, option)
+        traj[idx + 1] = next_state
+        # Need to make a new option without policy, initiable, and
```

Comment:
did you verify that pickling actually fails without this? i'm a little surprised

Reply:
maybe it's because the termination function or policy uses behavior classifiers internally?

Reply:
Yes and Yes